packages/host/app/lib/mutex.ts

```diff
@@ -0,0 +1,43 @@
+import { buildWaiter } from '@ember/test-waiters';
+
+/* Usage example:
+
+// Usage example
+const mutex = new Mutex();
+
+async function criticalSection() {
+  await mutex.dispatch(async () => {
+    // Your critical section code here
+    console.log('Critical section');
+  });
+}
+*/
+
```

Comment:
maybe provide some more description on what this is doing and when you'd use it. I think this makes it so that the critical section is only ever executed once at a time. is that right?

Reply:
Updated!